### PR TITLE
Breaking: [AEA-4545] - Move to Syft

### PR DESCRIPTION
**Pull request by wildjames**
## Summary

- :exclamation: Breaking Change
- :robot: Operational or Infrastructure Change

### Details
We need to use syft, rather than CycloneDX.

While I was at it, I rewrote the action to be a bit more portable. Since we always use dev containers, I leverage those rather than using a dedicated SBOM container. The new version builds and runs the SBOM scripts inside the devcontainer defined for the repository, and uploads the artifacts. Usage is also significantly easier:

```
name: SBOM scan PR

on:
  pull_request:
    branches: [main]

jobs:
  sbom_scans:
    uses: NHSDigital/eps-action-sbom/.github/workflows/sbom_workflow.yml@v2.0.0
```

When this is merged, I'll release a `v2.0.0` tag, and update the quality-checks repo.

Example of this running: https://github.com/NHSDigital/electronic-prescription-service-clinical-prescription-tracker/actions/runs/11277396474/job/31363438893

### CODE REVIEW

File: run-sbom-scan.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+for sbom in ./sbom*.json; do
+  if [ -s "$sbom" ]; then
+    echo "$sbom exists and has data. Scanning with Grype..."
+    grype sbom:"$sbom" -o json > "$(basename "$sbom" .json)-analysis.json"
+    echo "Done"
+  fi
+done
+
+# Download the check script
+curl -sSfL https://raw.githubusercontent.com/NHSDigital/eps-action-sbom/refs/heads/aea-0000-move-to-syft/check-sbom-issues-against-ignores.sh -o check-sbom-issues-against-ignores.sh
```
Comment (anthony-nhs) [inline run-sbom-scan.sh:57]: this references your current branch. This should be a tag instead (which I know makes it hard to test...)

File: run-sbom-scan.sh [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+fi
+
+# Scan each SBOM with Grype
+for sbom in ./sbom*.json; do
+  if [ -s "$sbom" ]; then
+    echo "$sbom exists and has data. Scanning with Grype..."
+    grype sbom:"$sbom" -o json > "$(basename "$sbom" .json)-analysis.json"
+    echo "Done"
+  fi
+done
```
Comment (anthony-nhs) [inline run-sbom-scan.sh:54]: can we add golang in here (for https://github.com/NHSDigital/electronic-prescription-service-get-secrets)

File: run-sbom-scan.sh (hunk 1 above)
Comment (wildjames) [reply run-sbom-scan.sh:57]: This is the tag I plan on releasing on a merge.

File: entrypoint.sh [hunk 3]
```diff
@@ -1,78 +1,170 @@
-#!/bin/sh -l
+#!/bin/bash -l
```
Comment (anthony-nhs) [inline entrypoint.sh:1]: if we are doing it really properly it should be `#!/usr/bin/env bash` to make sure it uses the version of bash installed in an environment (which may be different to the one in /bin/bash)

File: entrypoint.sh [hunk 4]
```diff
@@ -1,78 +1,170 @@
...
-    echo "ASDF_DIR not set. Copying in local installation of asdf..."
-    mkdir -p "${ASDF_DIR}"
-    cp -r /root/.asdf/* "${ASDF_DIR}"/
+    echo "ASDF_DIR has been provided: ${ASDF_DIR}"
+else
+  export ASDF_DIR="/root/.asdf/"
 fi
+export PATH="$PATH:$ASDF_DIR/bin:$ASDF_DIR/shims"
 asdf reshim
+asdf install
```
Comment (anthony-nhs) [inline entrypoint.sh:19]: install should be before reshim (reshim fixes the synmlinks created by an install)

**Review by jonathanwelch1-nhs — APPROVED** [review]

**Pull request merged**